src/pages/home/sidebar/SidebarScreen/FloatingActionButtonAndPopover.js

```diff
@@ -146,19 +151,17 @@ function FloatingActionButtonAndPopover(props) {
         }
     };
 
-    useEffect(
-        () => {
-            const navigationState = props.navigation.getState();
-            const routes = lodashGet(navigationState, 'routes', []);
-            const currentRoute = routes[navigationState.index];
-            if (currentRoute && ![NAVIGATORS.CENTRAL_PANE_NAVIGATOR, SCREENS.HOME].includes(currentRoute.name)) {
-                return;
-            }
+    useEffect(() => {
+        const navigationState = props.navigation.getState();
+        const routes = lodashGet(navigationState, 'routes', []);
+        const currentRoute = routes[navigationState.index];
+        if (currentRoute && ![NAVIGATORS.CENTRAL_PANE_NAVIGATOR, SCREENS.HOME].includes(currentRoute.name)) {
+            return;
+        }
+        if (!props.showDownloadAppBanner || !Browser.isMobile()) {
             Welcome.show({routes, showCreateMenu});
-        },
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-        [],
-    );
+        }
+    }, [props.showDownloadAppBanner, props.navigation, showCreateMenu]);
```

Comment:
Why are we now running this function whenever these dependencies change vs only when the component mounts?

Reply:
We want to run this when `props.showDownloadAppBanner` changes

Reply:
Ah ok that makes sense. It will also run when props.navigation changes. I don't know exactly when that happens but I imagine it updates fairly often. I think we should make sure that Welcome.show isn't triggered multiple times.

But now I see that we have an early return at the top of that function so I think it's good.https://github.com/Expensify/App/blob/7d1f5c64168d0bb70c204876a1da7ffa447e96d4/src/libs/actions/Welcome.js#L109-L111